internal/config/config_test.go

```diff
@@ -116,64 +278,231 @@ func TestGlobalConfig_UnmarshalJSON(t *testing.T) {
+				"logging": {
+					"logger": "glg",
+					"level": "warn",
+					"format": "json"
+				}
+			}`)
+			return test{
+				name: "return nil when time_zone key is empty and json unmarshal successes",
+				args: args{
+					data: data,
+				},
+				fields: fields{},
+				want: want{
+					want: &GlobalConfig{
+						Version: "v1.0.0",
+						Logging: &Logging{
+							Logger: "glg",
+							Level:  "warn",
+							Format: "json",
+						},
+					},
+					err: nil,
+				},
+			}
+		}(),
+		func() test {
+			data := []byte(`{
+				"version": "v1.0.0",
+				"time_zone": "UTC"
+			}`)
+			return test{
+				name: "return nil when logging key is empty and json unmarshal successes",
+				args: args{
+					data: data,
+				},
+				fields: fields{},
+				want: want{
+					want: &GlobalConfig{
+						Version: "v1.0.0",
+						TZ:      "UTC",
+					},
+					err: nil,
+				},
+			}
+		}(),
+		func() test {
+			data := []byte(`{
+				"version": "v1.0.0",
+				"time_zone": "UTC",
+				"logging": {
+					"level": "warn",
+					"format": "json"
+				}
+			}`)
+			return test{
+				name: "return nil when logging.logger key is empty and json unmarshal successes",
+				args: args{
+					data: data,
+				},
+				fields: fields{},
+				want: want{
+					want: &GlobalConfig{
+						Version: "v1.0.0",
+						TZ:      "UTC",
+						Logging: &Logging{
+							Level:  "warn",
+							Format: "json",
+						},
+					},
+					err: nil,
+				},
+			}
+		}(),
+		func() test {
+			data := []byte(`{
+				"version": "v1.0.0",
+				"time_zone": "UTC",
+				"logging": {
+					"logger": "glg",
+					"format": "json"
+				}
+			}`)
+			return test{
+				name: "return nil when logging.level key is empty and json unmarshal successes",
+				args: args{
+					data: data,
+				},
+				fields: fields{},
+				want: want{
+					want: &GlobalConfig{
+						Version: "v1.0.0",
+						TZ:      "UTC",
+						Logging: &Logging{
+							Logger: "glg",
+							Format: "json",
+						},
+					},
+					err: nil,
+				},
+			}
+		}(),
+		func() test {
+			data := []byte(`{
+				"version": "v1.0.0",
+				"time_zone": "UTC",
+				"logging": {
+					"logger": "glg",
+					"level": "warn"
+				}
+			}`)
+			return test{
+				name: "return nil when logging.format key is empty and json unmarshal successes",
+				args: args{
+					data: data,
+				},
+				fields: fields{},
+				want: want{
+					want: &GlobalConfig{
+						Version: "v1.0.0",
+						TZ:      "UTC",
+						Logging: &Logging{
+							Logger: "glg",
+							Level:  "warn",
+						},
+					},
+					err: nil,
+				},
+			}
+		}(),
+		func() test {
+			data := []byte(`{time_zone}`)
+			return test{
+				name: "return unmarshal error when json data is invalid",
+				args: args{
+					data: data,
+				},
+				fields: fields{},
+				want: want{
+					want: &GlobalConfig{},
+					err:  errors.New("readFieldHash: expect \", but found t, error found in #2 byte of ...|{time_zone}|..., bigger context ...|{time_zone}|..."),
+				},
+			}
+		}(),
```

Comment:
- how about adding the following cases?
1. `data = []byte{}`
2. `data == nil`

Reply:
thank you for your comment.
I agree with you :bow: